src/main/java/eu/dissco/backend/repository/ElasticSearchRepository.java

```diff
@@ -37,94 +35,50 @@ public List<DigitalSpecimen> search(String query, int pageNumber, int pageSize)
         .q("_exists_:digitalSpecimen.physicalSpecimenId AND " + query).from(offset).size(pageSize)
         .build();
     return client.search(searchRequest, ObjectNode.class).hits().hits().stream().map(Hit::source)
-        .map(this::mapToDigitalSpecimen).toList();
+        .map(this::mapDigitalSpecimenToJsonApiData).toList();
   }
 
-  public List<DigitalSpecimen> getLatestSpecimen(int pageNumber, int pageSize) throws IOException {
+  public List<JsonApiData> getLatestSpecimen(int pageNumber, int pageSize) throws IOException {
     var offset = getOffset(pageNumber, pageSize);
     var searchRequest = new SearchRequest.Builder().index(INDEX)
         .q("_exists_:digitalSpecimen.physicalSpecimenId ")
         .sort(s -> s.field(f -> f.field(FIELD_CREATED).order(SortOrder.Desc))).from(offset)
         .size(pageSize).build();
     return client.search(searchRequest, ObjectNode.class).hits().hits().stream().map(Hit::source)
-        .map(this::mapToDigitalSpecimen).toList();
+        .map(this::mapDigitalSpecimenToJsonApiData).toList();
   }
 
-  public List<AnnotationResponse> getLatestAnnotations(int pageNumber, int pageSize)
+  public List<JsonApiData> getLatestAnnotations(int pageNumber, int pageSize)
       throws IOException {
     var offset = getOffset(pageNumber, pageSize);
 
     var searchRequest = new SearchRequest.Builder().index(INDEX).q(ANNOTATION_EXISTS_QUERY)
         .sort(s -> s.field(f -> f.field(FIELD_CREATED).order(SortOrder.Desc))).from(offset)
         .size(pageSize).build();
     return client.search(searchRequest, ObjectNode.class).hits().hits().stream().map(Hit::source)
-        .map(this::mapToAnnotationResponse).toList();
+        .map(this::mapAnnotationToJsonApiData).toList();
   }
 
-  public List<JsonApiData> getLatestAnnotationsJsonResponse(int pageNumber, int pageSize)
-      throws IOException {
-    var offset = getOffset(pageNumber, pageSize);
-
-    var searchRequest = new SearchRequest.Builder().index(INDEX).q(ANNOTATION_EXISTS_QUERY)
-        .sort(s -> s.field(f -> f.field(FIELD_CREATED).order(SortOrder.Desc))).from(offset)
-        .size(pageSize).build();
-    return client.search(searchRequest, ObjectNode.class).hits().hits().stream().map(Hit::source)
-        .map(this::mapToJsonApiData).toList();
-  }
-
-  public List<AnnotationResponse> getLatestAnnotation() throws IOException {
-    var searchRequest = new SearchRequest.Builder().index(INDEX).q(ANNOTATION_EXISTS_QUERY)
-        .sort(s -> s.field(f -> f.field(FIELD_CREATED).order(SortOrder.Desc))).size(10).build();
-    return client.search(searchRequest, ObjectNode.class).hits().hits().stream().map(Hit::source)
-        .map(this::mapToAnnotationResponse).toList();
+  private JsonApiData mapDigitalSpecimenToJsonApiData(ObjectNode json){
+    var digitalSpecimen = (ObjectNode) json.get("digitalSpecimen");
+    return  new JsonApiData(digitalSpecimen.get("id").asText(), digitalSpecimen.get("type").asText(), digitalSpecimen );
```

Comment:
What worries me a bit is that we take the object as-is over. We might want to map to an object first and then put it in attributes, then at least we are sure we serve the same object from database or elastic

Reply:
This has been fixed